### PR TITLE
WINC-511: Use RuntimeClass in e2e tests

### PR DESCRIPTION
**Pull request by sebsoto**
Documents the use of a RuntimeClass when deploying Windows workloads, and uses them within the e2e tests.

### CODE REVIEW

File: docs/windows-workloads.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    node.kubernetes.io/windows-build: '10.0.17763'
+  tolerations:
+    - effect: NoSchedule
+      key: os
+      operator: Equal
+      value: "windows"
+    - effect: NoSchedule
+      key: os
+      operator: Equal
+      value: "Windows"
```
Comment (jrvaldes) [inline docs/windows-workloads.md:40]: > K8s documentation suggests using lowercase "windows", but WMCO registers nodes with uppercase "Windows".

Is this relevant here as well?

**Review by jrvaldes — COMMENTED** [review]
@sebsoto  Thanks for working on this. Left minor suggestions. PTAL.

Wondering if the  windows-workloads.md should be linked in the README.md

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

I guess using RuntimeClass [doesn't work for DaemonSets](https://github.com/kubernetes/kubernetes/issues/111590) (yet?)

**Comment by mansikulkarni96** [comment]
/approve
Thanks for working on this @sebsoto

File: docs/windows-workloads.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        image: mcr.microsoft.com/windows/servercore:ltsc2019
+        imagePullPolicy: IfNotPresent
+        command:
+        - powershell.exe
+        - -command
+        - $listener = New-Object System.Net.HttpListener; $listener.Prefixes.Add('http://*:80/'); $listener.Start();Write-Host('Listening at http://*:80/'); while ($listener.IsListening) { $context = $listener.GetContext(); $response = $context.Response; $content='<html><body><H1>Red Hat OpenShift + Windows Container Workloads</H1></body></html>'; $buffer = [System.Text.Encoding]::UTF8.GetBytes($content); $response.ContentLength64 = $buffer.Length; $response.OutputStream.Write($buffer, 0, $buffer.Length); $response.Close(); };
+        securityContext:
+          runAsNonRoot: false
+          windowsOptions:
+            runAsUserName: "ContainerAdministrator"
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply docs/windows-workloads.md:74]: For this specific workload I think it is, as permissions are required to create the webserver. This is just being copied from docs/custom-workload.md

**Pull request merged**